feat: Add `transcript_junction_builder` module

jarbesfeld opened the pull request:
closes #344 and #346

This notebook demonstrates how an `AssayedFusion` object can be generated for a toy BCR::ABL1 fusion example that comes from the literature using older transcript versions.

korikuzma reviewed (commented):
Alex to review

jarbesfeld commented:
@ahwagner I updated this PR to include a "quick start" module that would allow newcomers to readily create FUSOR objects. As a note, this currently only supports fusion junctions that are represented using RefSeq, HGVS transcript accessions and coordinates.

@korikuzma I think this PR should be ready for your review, in particular the `quick_start.py` and `test_quick_start.py` files.

korikuzma commented on src/fusor/transcript_junction_builder.py:
```diff
@@ -0,0 +1,145 @@
+    async def _get_junc_location(
+        self, tx: str, pos: int, cds_start: int = 0
+    ) -> GenomicTxMetadata:
+        """Get junction data from transcript
```
I would make a comment that the target assembly will always use GRCh38

jarbesfeld commented:
@korikuzma I discussed with @ahwagner and he had a few requests:

1. Clarify what "residue" means in constructor docstrings. I added a note clarifying that these describe 1-based coordinates on a transcript or genomic sequence.
2. Add support for genomic breakpoints that are described in intronic regions. I added support for this feature along with tests as well as a example in the jupyter notebook.
3. Add support for linker sequences.

Alex had said that the structure of the module was improved, so this should be ready for your review
